File: src/main/java/pl/lodz/p/carpooling/search/SearchController.java
package pl.lodz.p.carpooling.search;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import pl.lodz.p.carpooling.transit.Transit;

import java.util.List;

/**
 * Created by Mateusz Surmanski on 29.11.15.
 */
@RestController
public class SearchController {

    @Autowired
    private SearchService searchService;

    @RequestMapping(value = "/searchByRouteAndDate", method = RequestMethod.GET)
    public ResponseEntity findTransits(@RequestParam String startCity, @RequestParam String endCity, @RequestParam String date, @RequestParam String time) {
        try {
            List<Transit> transits = searchService.findTransits(startCity,endCity,date,time);
            return ResponseEntity.ok(transits);
        } catch (Exception e) {
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
        }
    }

    @RequestMapping(value = "/searchByRoute", method = RequestMethod.GET)
    public ResponseEntity findTransitsByRoute(@RequestParam String startCity, @RequestParam String endCity) {
        try {
            List<Transit> transits = searchService.findTransits(startCity,endCity);
            return ResponseEntity.ok(transits);
        } catch (Exception e) {
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
        }
    }
}

File: src/main/resources/public/app/search/search.controller.js
angular.module('carpooling.search', []).controller('SearchController', ['$scope', '$rootScope', '$http', '$state',
    function ($scope, $rootScope, $http, $state) {

        $scope.search = function (searchObject) {
            if(searchObject.date === null || searchObject.date === undefined) {
                searchByRoute(searchObject);
            } else {
                searchByRouteAndDate(searchObject);
            }
        };


        function searchByRouteAndDate(searchObject) {
            $http({
                url: '/searchByRouteAndDate',
                method: 'GET',
                params: {
                    'startCity': searchObject.startCity,
                    'endCity': searchObject.endCity,
                    'date': searchObject.date,
                    'time': searchObject.time
                }
            }).success(function (response) {
                $rootScope.results = response;
                $state.go('search.results');
            }).error(function (error) {
                console.error('search error');
            });
        }

        function searchByRoute(searchObject) {
            $http({
                url: '/searchByRoute',
                method: 'GET',
                params: {
                    'startCity': searchObject.startCity,
                    'endCity': searchObject.endCity
                }
            }).success(function (response) {
                $rootScope.results = response;
                $state.go('search.results');
            }).error(function (error) {
                console.error('search error');
            });
        }
    }]);